library/std/src/primitive_docs.rs

```diff
@@ -194,14 +194,47 @@ mod prim_bool {}
 /// # `!` and traits
 ///
 /// When writing your own traits, `!` should have an `impl` whenever there is an obvious `impl`
-/// which doesn't `panic!`. As it turns out, most traits can have an `impl` for `!`. Take [`Debug`]
+/// which doesn't `panic!`. The reason is that functions returning an `impl Trait` cannot have
+/// divergence, i.e., returning `!`, as their only possible code path. As an example, this code
```

Comment:
This is only true _if `!` does not implement the trait_. Which is sort of the whole point of this documentation :P

Reply:
nit: 'cannot have divergence' -> 'cannot diverge'